src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.OSX.cs

```diff
@@ -10,6 +10,7 @@ namespace System.Diagnostics
     public partial class Process
     {
         private const int NanosecondsTo100NanosecondsFactor = 100;
+        private static readonly Interop.libSystem.mach_timebase_info_data_t timeBase = GetTimeBase();
```

Comment:
It would be better to initialize this field lazily, when we need it for the first time. Otherwise this sys-call:
- may be called even if we don't need it (during type initializaiton)
- in theory it may fail and be very hard to handle properly

Since we use the following logic in 4 places:

```cs
return new TimeSpan(Convert.ToInt64($ulong * timeBase.numer / timeBase.denom / NanosecondsTo100NanosecondsFactor));
```

 we could introduce a helper method that could take care of everything, something like this:

```suggestion
        private static volatile uint s_timeBase_numer, s_timeBase_denom;
        
        private static TimeSpan Map(ulong sysTime)
        {
            uint denom = s_timeBase_denom;
            if (denom == default)
            {
                Interop.libSystem.mach_timebase_info_data_t timeBase = GetTimeBase();
                s_timeBase_denom = denom = timeBase.denom;
                s_timeBase_numer = timeBase.numer;
            }
            uint numer = s_timeBase_numer;
            
            return new TimeSpan(Convert.ToInt64(sysTime * numer / denom / NanosecondsTo100NanosecondsFactor));
        }
```

Reply:
Yeah, agree with your reasoning and applied a bit modified version of this snippet (the only change is in naming, I like `MapTime` a little bit better).